sdk/face/ai-vision-face-rest/samples-dev/personDirectory.ts

```diff
@@ -0,0 +1,318 @@
+    IdentifyFromPersonDirectory200Response,
+    UpdateDynamicPersonGroupWithPersonChanges202Response,
+    UpdateDynamicPersonGroupWithPersonChangesBodyParam,
+    VerifyFromPersonDirectory200Response,
+    getLongRunningPoller,
+} from '@azure-rest/ai-vision-face';
+
+/**
+ * This sample demonstrates how to create a liveness detection session.
+ *
+ * @summary creates a liveness detection session
+ */
+
+interface PersonData {
+    name: string;
+    userData: string;
+}
+
+interface FaceData {
+    fileName: string;
+    userData: string;
+}
+
+interface CreatePersonWithFacesResponse {
+    person: CreatePerson202Response;
+    faces: AddPersonFace202Response[];
+}
+
+interface IdentifyRequestBody {
+    dynamicPersonGroupId?: string;
+    personIds?: string[];
+}
+
+const poll = async (client: FaceClient, initialResponse: HttpResponse): Promise<void> => {
+    const poller = await getLongRunningPoller(client, initialResponse as any);
+    await poller.pollUntilDone();
+    console.log(`Long running operation: ${poller.getOperationState().status}`);
+};
+
+const createPersonWithFaces = async (client: FaceClient, personData: PersonData, faceData: FaceData[]): Promise<CreatePersonWithFacesResponse> => {
+    const createPerson = async (data: PersonData): Promise<CreatePerson202Response> => {
+        const { name, userData } = data;
+        const response = await client.path('/persons').post({
+            body: { name, userData },
+        }) as CreatePerson202Response;
+        console.log('Create person:');
+        console.log(JSON.stringify(response.body, null, 2));
+        return response;
+    };
+
+    const addPersonFace = async (personId: string, data: FaceData): Promise<AddPersonFace202Response> => {
+        const { fileName, userData } = data;
+        const response = await client.path('/persons/{personId}/recognitionModels/{recognitionModel}/persistedfaces', personId, 'recognition_04').post({
+            contentType: 'application/octet-stream',
+            queryParameters: { userData },
+            body: readFileSync(`samples-dev/data/${fileName}`),
+        }) as AddPersonFace202Response;
+        console.log(`Add person face: ${fileName}`);
+        console.log(JSON.stringify(response.body, null, 2));
+        return response;
+    };
+
+    const person = {} as CreatePersonWithFacesResponse;
+    person.person = await createPerson(personData);
+    // You can directly add faces to the person before the Person async operation is completed.
+    person.faces = await Promise.all(faceData.map(data => addPersonFace(person.person.body.personId, data)));
+    return person;
+};
+
+const deletePerson = async (client: FaceClient, personId: string): Promise<void> => {
+    const response = await client.path('/persons/{personId}', personId).delete() as DeletePerson202Response;
+    console.log('Delete person:');
+    console.log(JSON.stringify(response.body, null, 2));
+    await poll(client, response);
+};
```

Comment:
I am not sure if we should encourage to wrap a few of the rest api calls into a function as a user experience sample for RLC @joheredi any thoughts?

Reply:
Another point is, if this is really what we want our customers to use face api, maybe wait until modular is ready is a better choice?

Reply:
Agree with @qiaozha, we should keep this simple and not try to abstract into higher level functions

Reply:
What would you suggest for reuse of the code?

Reply:
For RLC, reuse of the code should be a non-goal I guess, @joheredi could help doubt confirm.